### DISCUSSION TITLE
Anonymous alias is not displayed in children but IS in fields. Bug or feature?

### DISCUSSION THREAD

**Discussion by B1Z0N** [tree-sitter/tree-sitter#2568] [Q&A]
When you have [anonymous](https://tree-sitter.github.io/tree-sitter/creating-parsers#:~:text=appear%20as%20an-,anonymous%20node,-%2C%20as%20if%20the) alias node like this: 

```js
myrule: $ => alias('HelloWorld', 'myhelloworld'),
``` 
It won't display itself in `children` in `src/node-types.json` (types are empty):

```json
{
  "type": "myrule",
  "named": true,
  "children": {
    "multiple": false,
    "required": true,
    "types": []
  }
}
``` 

But once you have it like this:

```js
myrule: $ => field('myfield', alias('HelloWorld', 'myhelloworld')),
``` 
It starts to appear in the output:

```json
{
  "type": "myrule",
  "named": true,
  "fields": {
    "myfield": {
      "multiple": false,
      "required": true,
      "types": [
        {
          "type": "myhelloworld",
          "named": false
        }
      ]
    }
  }
}
```

What is the reason behind this design? Or is it a bug?

**Comment by amaanq** [comment]
a single terminal just makes the rule terminal and absorbs the terminal symbol.

Any fields, choices, sequences or others that cause it to not be terminal will make the data visible in node_types.json

Now I'm not sure if field alone should make the rule non terminal though, but it is probably done so queries don't break.

**Comment by ahlinc** [comment]
I think it's a bug in the first case.

But overall re aliasing a one anonymous terminal to an another one in most cases is an anti pattern, this aliasing to a anonymous terminals mechanism exists for example for possibility to alias an external tokens that represent static strings.
